Mcu/l431/Src/WS2812.c

```diff
@@ -0,0 +1,61 @@
+#include "WS2812.h"
+
+#include "targets.h"
+
+#ifdef USE_LED_STRIP
+
+#ifndef WS2812_PORT
+    #error "WS2812_PORT is not defined"
+#endif
+
+#ifndef WS2812_PIN
+    #error "WS2812_PIN is not defined"
+#endif
+
+void enableDWT_CycleCounter(void)
+{
+    // Enable TRCENA in the DEMCR (Debug Exception and Monitor Control Register)
+    CoreDebug->DEMCR |= CoreDebug_DEMCR_TRCENA_Msk;
+    // Enable the cycle counter
+    DWT->CTRL |= DWT_CTRL_CYCCNTENA_Msk;
+    // Reset the cycle counter
+    DWT->CYCCNT = 0;
+}
+
+void waitClockCycles(uint16_t cycles)
+{
+    DWT->CYCCNT = 0; // Reset the cycle counter
+    while (DWT->CYCCNT < cycles) {
+        // Wait until the specified number of cycles has passed
+    }
+}
+
+void sendBit(uint8_t inbit)
+{
+    WS2812_PORT->BSRR = WS2812_PIN;
+    waitClockCycles(CPU_FREQUENCY_MHZ >> (2 - inbit));
+    WS2812_PORT->BRR = WS2812_PIN;
+    waitClockCycles(CPU_FREQUENCY_MHZ >> (1 + inbit));
+}
+
+void send_LED_RGB(uint8_t red, uint8_t green, uint8_t blue)
+{
+    __disable_irq();
+    uint32_t twenty_four_bit_color_number = green << 16 | red << 8 | blue;
```

Comment:
should we enable/disable the DWT in this function? any overhead to keeping it enabled all the time?

Reply:
updated, now it enable/disable in every call of send_LED_RGB